lib/service/storage/elasticsearch.js

```diff
@@ -2362,10 +2362,49 @@ class ElasticSearch extends Service {
 
     return parsedValue;
   }
+
+  /**
+   * Returns true if one of the mappings dynamic property changes value from
+   * false to true
+   */
+  _dynamicChanges (previousMappings, newMappings) {
+    const previousValues = findDynamic(previousMappings);
+
+    for (const [path, previousValue] of Object.entries(previousValues)) {
+      if (previousValue.toString() !== 'false') {
+        continue;
+      }
+
+      const newValue = _.get(newMappings, path);
+
+      if (newValue && newValue.toString() === 'true') {
+        return true;
+      }
+    }
+
+    return false;
+  }
 }
 
 module.exports = ElasticSearch;
 
+/**
+ * Finds paths and values of mappings dynamic properties
+ */
+function findDynamic (mappings, path = [], results = {}) {
+  if (mappings.dynamic !== undefined) {
+    results[path.concat('dynamic').join('.')] = mappings.dynamic;
+  }
+
+  for (const [key, value] of Object.entries(mappings)) {
+    if (_.isPlainObject(value)) {
```

Comment:
(nitpicking) you shoud use our own `isPlainObject` method from `util/safeObject.js`